docs/cluster_class.md

```diff
@@ -0,0 +1,223 @@
+    pods:
+      cidrBlocks:
+        - 192.168.0.0/16
+    serviceDomain: cluster.local
+  topology:
+    class: quickstart
+    version: ${KUBERNETES_VERSION}
+    controlPlane:
+      metadata: {}
+      replicas: 1
+    variables:
+      - name: bundleLookupBaseRegistry
+        value: "projects.registry.vmware.com/cluster_api_provider_bringyourownhost"
+      - name: controlPlaneIpAddr
+        value: ${CONTROL_PLANE_ENDPOINT_IP}
+      - name: kubeVipPodManifest
+        value: |
+          apiVersion: v1
+          kind: Pod
+          metadata:
+            creationTimestamp: null
+            name: kube-vip
+            namespace: kube-system
+          spec:
+            containers:
+            - args:
+              - manager
+              env:
+              - name: cp_enable
+                value: "true"
+              - name: vip_arp
+                value: "true"
+              - name: vip_leaderelection
+                value: "true"
+              - name: vip_address
+                value: ${CONTROL_PLANE_ENDPOINT_IP}
+              - name: vip_interface
+                value: "{{ .DefaultNetworkInterfaceName }}"
+              - name: vip_leaseduration
+                value: "15"
+              - name: vip_renewdeadline
+                value: "10"
+              - name: vip_retryperiod
+                value: "2"
+              image: ghcr.io/kube-vip/kube-vip:v0.4.1
+              imagePullPolicy: IfNotPresent
+              name: kube-vip
+              resources: {}
+              securityContext:
+                capabilities:
+                  add:
+                  - NET_ADMIN
+                  - NET_RAW
+              volumeMounts:
+              - mountPath: /etc/kubernetes/admin.conf
+                name: kubeconfig
+            hostNetwork: true
+            hostAliases:
+              - hostnames:
+                  - kubernetes
+                ip: 127.0.0.1
+            volumes:
+            - hostPath:
+                path: /etc/kubernetes/admin.conf
+                type: FileOrCreate
+              name: kubeconfig
+          status: {}
+    workers:
+      machineDeployments:
+        - class: quickstart-worker
+          metadata: {}
+          name: md-0
+          replicas: 1
+```
+
+### How to use with clusterctl?
+
+To create a cluster using clusterctl with managed topology, use `topology` flavor and set following environment variables :
+```
+CLUSTER_NAME
+CONTROL_PLANE_ENDPOINT_IP
+```
+Also set feature flag for ClusterClass:
+
+**Feature gate name:** `ClusterTopology`
+
+**Variable name to enable/disable the feature gate**: `CLUSTER_TOPOLOGY`
+
+After setting the variables use below command to create a yaml file to create a cluster with managed topology
+```shell
+clusterctl generate cluster <cluster-name> --kubernetes-version <kubernetes-version> --control-plane-machine-count=3 --worker-machine-count=3 \
+    --flavor topology > my-cluster.yaml
+```
+
+#### clusterctl alpha topology plan
```

Comment:
```suggestion
### clusterctl alpha topology plan
```
Just to make it a distinct heading

Reply:
Actually it's not a separate heading, and it's under clusterctl, so doesn't feel to make it distinct.